components/support/ffi/src/macros.rs

```diff
@@ -195,3 +195,37 @@ pub fn convert_to_json_string<T: serde::Serialize>(value: &T) -> *mut c_char {
     let as_string = serde_json::to_string(&value).unwrap();
     rust_string_to_c(as_string)
 }
+
+#[macro_export]
+macro_rules! impl_into_ffi_protobuf {
+    ($FFIType:ty) => {
+        unsafe impl $crate::IntoFfi for $FFIType {
+            type Value = $crate::ProtoWrapper;
+
+            fn ffi_default() -> Self::Value {
+                $crate::ProtoWrapper {
+                    data: ::std::ptr::null_mut(),
+                    len: 0,
+                }
+            }
+
+            fn into_ffi_value(self) -> Self::Value {
+                let bytes = self.write_to_bytes().unwrap();
+                let mut buf = bytes.into_boxed_slice();
```

Comment:
Also, I already told you this, but you need to `std::mem::forget(buf)` here. Sorry for not looking at this closer.

Reply:
And by here, I mean another line down after you're done using it >_>